test/libsolidity/GasCosts.cpp

```diff
@@ -82,6 +82,71 @@ BOOST_AUTO_TEST_CASE(string_storage)
 	}
 }
 
+
+// With no payable functions we run the callvalue check only once
+BOOST_AUTO_TEST_CASE(single_callvaluecheck_no_payable)
+{
+	char const* sourceCode = R"(
+		contract C {
+			address a;
+			function f(address b) public returns (address c) {
+				address d = b;
+				return d;
+			}
+			function f1(address b) public returns (address c) {
+				address d = b;
+				return d;
+			}
+			function f2(address b) public returns (address c) {
+				address d = b;
+				return d;
+			}
+			function f3(address b) public returns (address c) {
+				address d = b;
+				return d;
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+
```

Comment:
Perhaps move the comment from above the test here? The name of the test case already gives a hint what's been tested and with the comment the check would be become clearer.